tests/csm/rest/test_iam_users.py

```diff
@@ -4638,3 +4648,403 @@ def test_42272(self):
+        msg = resp_data[resp_msg_index]
+        self.log.info("Fetching internal IAM User")
+        resp_node = self.nd_obj.execute_cmd(cmd=common_cmd.K8S_GET_PODS,
+                                            read_lines=False,
+                                            exc=False)
+        pod_name = self.csm_obj.get_pod_name(resp_node)
+        self.log.info(pod_name)
+        self.nd_obj.execute_cmd(
+            cmd=common_cmd.K8S_CP_TO_LOCAL_CMD.format(
+                pod_name, self.cluster_conf_path, self.csm_copy_path, cons.CORTX_CSM_POD),
+            read_lines=False,
+            exc=False)
+        resp = self.nd_obj.copy_file_to_local(
+            remote_path=self.csm_copy_path, local_path=self.local_csm_path)
+        assert_utils.assert_true(resp[0], resp[1])
+        stream = open(self.local_csm_path, 'r')
+        data = yaml.load(stream, Loader=yaml.Loader)
+        internal_user = data["common"]["rgw"]["auth_user"]
+        payload = {"access_key":"sgiamadmin", "secret_key":"null"}
+        resp = self.csm_obj.modify_iam_user_rgw(internal_user, payload)
+        assert_utils.assert_true(resp.status_code == HTTPStatus.FORBIDDEN,
+                                 "Internal IAM User Modified")
+        if CSM_REST_CFG["msg_check"] == "enable":
+            self.log.info("Verifying error response...")  # TODO
+            assert_utils.assert_equals(resp.json()["error_code"], resp_error_code)
+            assert_utils.assert_equals(resp.json()["message_id"], resp_msg_id)
+            assert_utils.assert_equals(resp.json()["message"], msg)
+        self.log.info("##### Test completed -  %s #####", test_case_name)
+
+    @pytest.mark.lc
+    @pytest.mark.csmrest
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.parallel
+    @pytest.mark.tags('TEST-42289')
+    def test_42289(self):
+        """
+        Test that internal user should be visible on the GET IAM user list
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        test_cfg = self.csm_conf["test_42289"]
+        resp_error_code = test_cfg["error_code"]
+        resp_msg_id = test_cfg["message_id"]
+        resp_data = self.rest_resp_conf[resp_error_code][resp_msg_id]
+        resp_msg_index = test_cfg["message_index"]
+        msg = resp_data[resp_msg_index]
+        self.log.info("Fetching internal IAM User")
+        resp_node = self.nd_obj.execute_cmd(cmd=common_cmd.K8S_GET_PODS,
+                                            read_lines=False,
+                                            exc=False)
+        pod_name = self.csm_obj.get_pod_name(resp_node)
+        self.log.info(pod_name)
+        self.nd_obj.execute_cmd(
+            cmd=common_cmd.K8S_CP_TO_LOCAL_CMD.format(
+                pod_name, self.cluster_conf_path, self.csm_copy_path, cons.CORTX_CSM_POD),
+            read_lines=False,
+            exc=False)
+        resp = self.nd_obj.copy_file_to_local(
```

Comment:
Can u create function to get_internal_iam_user and just call it here

Reply:
done